extensions/functions_arithmetic.yaml

```diff
@@ -1614,3 +1614,131 @@ window_functions:
         decomposable: NONE
         return: i64?
         window_type: PARTITION
+  - name: "first_value"
+    description: >
+      The `expression` is evaluated against the first row.  The `expression` can be a
+      column, expression or subquery that evaluates to a single value.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "last_value"
+    description: >
+      The `expression` is evaluated against the last row.  The `expression` can be a
+      column, expression, or subquery that evaluates to a single value.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "nth_value"
+    description: >
+      The `expression` is evaluated against a row that comes after the current row based
+      on the `offset`.  `offset` should be a positive integer.  The `expression` can be a
+      column, expression or subquery that evaluates to a single value.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+          - value: i32
+            name: offset
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+      - args:
+          - value: any1
+            name: expression
+          - value: i64
+            name: offset
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "lead"
+    description: >
+      Return a value from a following row based on a specified physical offset.
+      This allows you to compare a value in the current row against a following row.
+
+      The `expression` is evaluated against a row that comes after the current row based
+      on the `offset`.  The `expression` can be a column, expression or subquery that
+      evaluates to a single value. The `offset` should be a positive integer and is set to
+      1 if not specified explicitly. The function returns the `default` input value if
```

Comment:
I don't see any `impls` that allow me to omit the offset (and I don't think there should be any).  Would it be more accurate to say the `offset` will be set to 1 if it is negative or null?  Is this how current engines work?

Reply:
Current engines will default to 1 if it's not set.  So it actually might be best to have an `impl` for that.

Reply:
Ok.  I think that will work.  We should still clarify what the behavior will be if the offset is -1 or null.

Reply:
Updated!  Also updated nth_value function to specify negative or null `window_offset` behavior.